src/solvers/tao_optimization_solver.C

```diff
@@ -398,7 +398,7 @@ extern "C"
     sys.update();
     X.swap(X_sys);
 
-    // Let's also wrap J and Jpre in PetscMatrix objects for convenience
+    // Let's also wrap J and Jpre in PetscMatrixBase objects for convenience
```

Comment:
Not sure if I missed these last time or if you got overzealous search-and-replacing comments to fix a converse complaint of mine last time?